ush/prep_sfc_snow.sh

```diff
@@ -0,0 +1,198 @@
+#------------------------------------------------------------------------
+
+MODEL_SNOW_FILE=${MODEL_SNOW_FILE:-"snogrb_model"}
+OUTPUT_GRIB2=${OUTPUT_GRIB2:-".false."}  # grib 1 when false.
+
+#------------------------------------------------------------------------
+# Do a quick check of the ims data to ensure it exists and is
+# not corrupt.
+#------------------------------------------------------------------------
+
+if [[ ! -f ${IMS_FILE} ]]; then
+  echo "WARNING: ${pgm} detects missing ims data. Will not run."
+  exit 7
+fi
+
+#------------------------------------------------------------------------
+# The model analysis time is set to the ims valid time, because the
+# ims data has highest priority of all input data.
+#------------------------------------------------------------------------
+
+${WGRIB2} -d 1 "${IMS_FILE}"
+err=$?
+if [[ ${err} -ne 0 ]]; then
+  echo "WARNING: ${pgm} detects corrupt ims data. Will not run."
+  exit 9
+else
+  tempdate=$(${WGRIB2} -t "${IMS_FILE}" | head -1) || true
+  IMSDATE=${tempdate#*d=}
+fi
+IMSDATE10=$(echo "${IMSDATE}" | cut -c1-10)
+IMSYEAR=$(echo "${IMSDATE10}" | cut -c1-4)
+IMSMONTH=$(echo "${IMSDATE10}" | cut -c5-6)
+IMSDAY=$(echo "${IMSDATE10}" | cut -c7-8)
+IMSHOUR=0   # emc convention is to use 00Z.
+
+#------------------------------------------------------------------------
+# Ensure AFWA data exists and is not too old.
+#------------------------------------------------------------------------
+
+if [[ ! -f ${AFWA_GLOBAL_FILE} ]]; then
+  echo "WARNING: ${pgm} detects missing afwa data. Will not run."
+  exit 3
+else
+  ${WGRIB2} -d 1 "${AFWA_GLOBAL_FILE}"
+  err=$?
+  if [[ ${err} -ne 0 ]]; then
+    echo "WARNING: ${pgm} detects corrupt afwa data. Will not run."
+    exit "${err}"
+  else
+    tempdate=$(${WGRIB2} -d 1 -t "${AFWA_GLOBAL_FILE}")
+    AFWADATE=${tempdate#*d=}
+    two_days_ago=$(${NDATE} -48 "${IMSDATE10}")
+    if [[ ${AFWADATE} -lt ${two_days_ago} ]]; then
+      echo "WARNING: ${pgm} detects old afwa data. Will not run."
+      exit 4
+    fi
+  fi
+fi
+
+pgmout=${pgmout:-"OUTPUT"}
+
+source prep_step
+
+rm -f ./fort.41
+cat > ./fort.41 << EOF
+ &source_data
+  autosnow_file=""
+  nesdis_snow_file="${IMS_FILE}"
+  nesdis_lsmask_file=""
+  afwa_snow_global_file="${AFWA_GLOBAL_FILE}"
+  afwa_snow_nh_file=""
+  afwa_snow_sh_file=""
+  afwa_lsmask_nh_file=""
+  afwa_lsmask_sh_file=""
+ /
+ &qc
+  climo_qc_file="${CLIMO_QC}"
+ /
+ &model_specs
+  model_lat_file="${MODEL_LATITUDE_FILE}"
+  model_lon_file="${MODEL_LONGITUDE_FILE}"
+  model_lsmask_file="${MODEL_SLMASK_FILE}"
+  gfs_lpl_file="${GFS_LONSPERLAT_FILE}"
+  /
+ &output_data
+  model_snow_file="./${MODEL_SNOW_FILE}"
+  output_grib2=${OUTPUT_GRIB2}
+ /
+ &output_grib_time
+  grib_year=${IMSYEAR}
+  grib_month=${IMSMONTH}
+  grib_day=${IMSDAY}
+  grib_hour=${IMSHOUR}
+ /
+ &parameters
+  lat_threshold=55.0
+  min_snow_depth=0.05
+  snow_cvr_threshold=50.0
+ /
+EOF
+
+eval "${SNOW2MDLEXEC}" >> "${pgmout}" 2> errfile
```

Comment:
Why is this `eval`?

Reply:
@GeorgeGayno-NOAA Why is this line using `eval`?

Reply:
> @GeorgeGayno-NOAA Why is this line using `eval`?

These scripts were first created many years ago. Maybe that was an NCO standard at the time? Feel free to remove it.